test/common/upstream/upstream_impl_test.cc

```diff
@@ -80,12 +80,80 @@ TEST(StrictDnsClusterImplTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized, ready());
-  EXPECT_CALL(dns_resolver, resolve("foo.bar.com", _))
-      .WillOnce(Invoke([&](const std::string&, Network::DnsResolver::ResolveCb cb)
-                           -> Network::ActiveDnsQuery* {
-                             cb(TestUtility::makeDnsResponse({"127.0.0.1", "127.0.0.2"}));
-                             return nullptr;
-                           }));
+  EXPECT_CALL(dns_resolver, resolve("foo.bar.com", Network::DnsLookupFamily::V4_ONLY, _))
+      .WillOnce(Invoke([&](const std::string&, const Network::DnsLookupFamily&,
+                           Network::DnsResolver::ResolveCb cb) -> Network::ActiveDnsQuery* {
+        cb(TestUtility::makeDnsResponse({"127.0.0.1", "127.0.0.2"}));
+        return nullptr;
+      }));
+  Json::ObjectSharedPtr loader = Json::Factory::loadFromString(json);
+  StrictDnsClusterImpl cluster(*loader, runtime, stats, ssl_context_manager, dns_resolver,
+                               dispatcher);
+  cluster.setInitializedCb([&]() -> void { initialized.ready(); });
+  EXPECT_EQ(2UL, cluster.hosts().size());
+  EXPECT_EQ(2UL, cluster.healthyHosts().size());
+}
+
+TEST(StrictDnsClusterImplTest, ImmediateResolveV6Only) {
+  Stats::IsolatedStoreImpl stats;
+  Ssl::MockContextManager ssl_context_manager;
+  NiceMock<Network::MockDnsResolver> dns_resolver;
+  NiceMock<Event::MockDispatcher> dispatcher;
+  NiceMock<Runtime::MockLoader> runtime;
+  ReadyWatcher initialized;
+
+  std::string json = R"EOF(
+  {
+    "name": "name",
+    "connect_timeout_ms": 250,
+    "type": "strict_dns",
+    "lb_type": "round_robin",
+    "hosts": [{"url": "tcp://foo.bar.com:443"}],
+    "dns_lookup_family" : "v6_only"
+  }
+  )EOF";
+
+  EXPECT_CALL(initialized, ready());
+  EXPECT_CALL(dns_resolver, resolve("foo.bar.com", Network::DnsLookupFamily::V6_ONLY, _))
+      .WillOnce(Invoke([&](const std::string&, const Network::DnsLookupFamily&,
+                           Network::DnsResolver::ResolveCb cb) -> Network::ActiveDnsQuery* {
+        cb(TestUtility::makeDnsResponse({"::1", "::2"}));
+        return nullptr;
+      }));
+  Json::ObjectSharedPtr loader = Json::Factory::loadFromString(json);
+  StrictDnsClusterImpl cluster(*loader, runtime, stats, ssl_context_manager, dns_resolver,
+                               dispatcher);
+  cluster.setInitializedCb([&]() -> void { initialized.ready(); });
+  EXPECT_EQ(2UL, cluster.hosts().size());
+  EXPECT_EQ(2UL, cluster.healthyHosts().size());
```

Comment:
Actually, I think for these bigger examples, `TEST_P` would make more sense. You can make the parameter a `std::tuple` of string for the `dns_lookup_family`, DNS family and the DNS response list.